.github/workflows/gosec.yaml

```diff
@@ -39,6 +39,7 @@ jobs:
           GOROOT: ""
         continue-on-error: true
       - name: Upload SARIF file
+        if: github.event_name != 'merge_group'
```

Comment:
you could use github.event_name == 'pull_request' instead, which is more explicit about when uploads should occur, but the current approach is equally valid. 
so I leave it to you to decide.